File: docs/workflows/scroll_pipeline.py
"""
Scroll Pipeline (Python)
TriadicFrameworks • Workflows Subsystem

Executes `.fff` scroll artifacts in Python environments.
Mirrors the JS pipeline API:

    run_scroll(scroll_text)

Returns:
    { "output": ..., "warnings": [...], "metadata": {...} }
"""

from tft.scrolls.parse import parse_scroll
from tft.scrolls.pipeline import execute_scroll


def run_scroll(scroll_text):
    """Parse and execute a scroll, returning a structured result."""
    parsed = parse_scroll(scroll_text)
    result = execute_scroll(parsed)

    return {
        "output": result.get("output"),
        "warnings": result.get("warnings", []),
        "metadata": result.get("metadata", {})
    }

File: docs/workflows/scrollPipeline.js
/**
 * Scroll Pipeline (JavaScript)
 * TriadicFrameworks • Workflows Subsystem
 *
 * Executes `.fff` scroll artifacts in browser environments.
 * Mirrors the Python pipeline API:
 *
 *     runScroll(scrollText)
 *
 * Returns:
 *     { output, warnings, metadata }
 */

import { parseScroll } from "./tft/scrolls/parse.js";
import { executeScroll } from "./tft/scrolls/pipeline.js";

export function runScroll(scrollText) {
    const parsed = parseScroll(scrollText);
    const result = executeScroll(parsed);

    return {
        output: result.output,
        warnings: result.warnings || [],
        metadata: result.metadata || {}
    };
}
